docs/spark/tutorials/ml-sentment-analysis.md

```diff
@@ -0,0 +1,207 @@
+
+1. Add the following additional `using` statements to the top of the *Program.cs*:
+
+   ```csharp
+   using System;
+   using System.Collections.Generic;
+   using Microsoft.ML;
+   using Microsoft.ML.Data;
+   using Microsoft.Spark.Sql;
+   using MLSparkModelML.Model;
+   ```
+
+1. Change the `DATA_FILEPATH` to the path of your *yelptest.csv*.
+
+1. Add the following code to your `Main` method to create a new `SparkSession`. The Spark Session is the entry point to programming Spark with the Dataset and DataFrame API.
+
+   ```csharp
+   SparkSession spark = SparkSession
+        .Builder()
+        .AppName(".NET for Apache Spark Sentiment Analysis")
+        .GetOrCreate();
+   ```
+
+   Calling the *spark* object created above allows you to access Spark and DataFrame functionality throughout your program.
+
+### Create a DataFrame and print to console
+
+Read in the Yelp review data from the *yelptest.csv* file as a `DataFrame`. Include `header` and `inferSchema` options. The `header` option reads the first line of *yelptest.csv* as column names instead of data. The `inferSchema` option infers column types based on the data.
+
+```csharp
+DataFrame df = spark
+    .ReadStream()
+    .Option("header", true)
+    .Option("inferSchema", true)
+    .Csv(DATA_FILEPATH);
+
+df.Show();
+```
+
+### Register a user-defined function
+
+You can use UDFs, *user-defined functions*, in Spark applications to do calculations and analysis on your data. In this tutorial, you use ML.NET with a UDF to evaluate each Yelp review.
+
+Add the following code to your `Main` method to register a UDF called `MLudf`.
+
+```csharp
+spark.Udf()
+    .Register<string, bool>("MLudf", predict);
+```
+
+This UDF takes a Yelp review string as input, and outputs true or false for positive or negative sentiments, respectively. It uses the *predict()* method that you define in a later step.
+
+### Use Spark SQL to call the UDF
+
+Now that you've read in your data and incorporated ML, use Spark SQL to call the UDF that will run sentiment analysis on each row of your DataFrame. Add the following code to your `Main` method:
+
+```csharp
+// Use Spark SQL to call ML.NET UDF
+// Display results of sentiment analysis on reviews
+df.CreateOrReplaceTempView("Reviews");
+DataFrame sqlDf = spark.Sql("SELECT ReviewText, MLudf(ReviewText) FROM Reviews");
+sqlDf.Show();
+
+// Print out first 20 rows of data
+// Prevent data getting cut off by setting truncate = 0
+sqlDf.Show(20, 0, false);
+
+spark.Stop();
+```
+
+### Create predict() method
+
+Add the following code before your `Main()` method. This code is similar to what is produced by Model Builder in *ConsumeModel.cs*. Moving this method to your console keeps the model loading every time you run your app.
+
+```csharp
+private static readonly PredictionEngine<ModelInput, ModelOutput> _predictionEngine;
+
+static Program()
```

Comment:
This took me a few times through to figure out that you were using a static constructor. I'd recommend adding a comment on that.

More importantly, it's very difficult to catch exceptions thrown in static constructors. That makes errors in them very hard to debug. For that reason, we usually recommend that static ctors are simple, with minimal logic. In this example, loading the zip file for the model could fail for a variety of reasons, like copying it to the wrong folder. Would that make it hard for readers to diagnose and fix? 

I'd consider refactoring this when you have time.